src/idpyoidc/server/session/manager.py

```diff
@@ -82,25 +78,33 @@ def ephemeral_id(*args, **kwargs):
     return uuid.uuid4().hex
 
 
-class SessionManager(Database):
+class SessionManager(GrantManager):
     parameter = Database.parameter.copy()
     # parameter.update({"salt": ""})
     init_args = ["handler"]
 
     def __init__(
-        self,
-        handler: TokenHandler,
-        conf: Optional[dict] = None,
-        sub_func: Optional[dict] = None,
-        remember_token: Optional[Callable] = None,
-        remove_inactive_token: Optional[bool] = False,
+            self,
+            handler: TokenHandler,
+            conf: Optional[dict] = None,
+            sub_func: Optional[dict] = None,
+            remember_token: Optional[Callable] = None,
+            remove_inactive_token: Optional[bool] = False,
     ):
         self.conf = conf or {"session_params": {"encrypter": default_crypt_config()}}
 
         session_params = self.conf.get("session_params") or {}
         _crypt_config = get_crypt_config(session_params)
         super(SessionManager, self).__init__(_crypt_config)
```

Comment:
Maybe `super(SessionManager, self).__init__(handler, _crypt_config)` ?

Reply:
Actually I think it's better to do super(SessionManager, self).__init__(handler, self.conf) after a check that `session_params` exists in `conf`.

Reply:
Actually we need **get_crypt_config**, it handles some backward compatibilities.
But I agree with the `super(SessionManager, self).__init__(handler, _crypt_config)` proposal

Reply:
Sorry to bring that back on the table, but there's a call of the parent constructor that expects a conf dict that contains the `session_params` key. We pass `_crypt_config` on the parent's constructor that in turn calls again the `get_crypt_config` with an empty dict.

Reply:
So I insist that super should be called with `self.conf`